fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetaClient.java

```diff
@@ -121,80 +126,83 @@ private AutoCloseClient getClient() throws MetaException {
         }
 
         synchronized (clientPoolLock) {
-            AutoCloseClient client = clientPool.poll();
+            RecyclableClient client = clientPool.poll();
             // The pool was empty so create a new client and return that.
             // Serialize client creation to defend against possible race conditions accessing
             // local Kerberos state
             if (client == null) {
-                return new AutoCloseClient(conf);
+                return new RecyclableClient(conf);
             } else {
                 return client;
             }
         }
     }
 
-    public List<String> getAllDatabaseNames() {
-        try (PlannerProfile.ScopedTimer ignored = PlannerProfile.getScopedTimer("HMS.getAllDatabases");
-                AutoCloseClient client = getClient()) {
-            return client.hiveClient.getAllDatabases();
+    public <T> T callRPC(String methodName, String messageIfError, Object... args) {
+        RecyclableClient client = null;
+        StarRocksConnectorException connectionException = null;
+        Class<?>[] argTypes = new Class[args.length];
+        for (int i = 0; i < args.length; i++) {
+            argTypes[i] = args[i].getClass();
+        }
+
+        try {
+            client = getClient();
+            Method method = client.hiveClient.getClass().getDeclaredMethod(methodName, argTypes);
+            return (T) method.invoke(client.hiveClient, args);
         } catch (Exception e) {
-            LOG.error("Failed to get all database names", e);
-            throw new StarRocksConnectorException("Failed to get all database names from meta store: " + e.getMessage());
+            LOG.error(messageIfError, e);
+            connectionException = new StarRocksConnectorException(messageIfError + ", msg: " + e.getMessage());
+            throw connectionException;
+        } finally {
+            if (client == null && connectionException != null) {
```

Comment:
when does the `client` will be `null`.

Reply:
getClient() failed. it also failed to init HiveMetaStoreThriftClient.